server/routing/routing.go

```diff
@@ -47,19 +57,22 @@ func New(ctx context.Context, store types.StoreAPI, opts types.APIOptions) (type
 }
 
 func (r *route) Publish(ctx context.Context, ref *corev1.RecordRef, record *corev1.Record) error {
-	// always publish data locally for archival/querying
+	// Always publish data locally for archival/querying
 	err := r.local.Publish(ctx, ref, record)
 	if err != nil {
 		st := status.Convert(err)
 
 		return status.Errorf(st.Code(), "failed to publish locally: %s", st.Message())
 	}
 
-	err = r.remote.Publish(ctx, ref, record)
-	if err != nil {
-		st := status.Convert(err)
+	// Only publish to network if peers are available
+	if r.hasPeersInRoutingTable() {
+		err = r.remote.Publish(ctx, ref, record)
+		if err != nil {
+			st := status.Convert(err)
 
-		return status.Errorf(st.Code(), "failed to publish to the network: %s", st.Message())
+			return status.Errorf(st.Code(), "failed to publish to the network: %s", st.Message())
+		}
 	}
```

Comment:
we dont necessarily have to have any peers to enable publication. once peers connect, they can negotiate if they want to sync their routing tables to check availability